patches/tModLoader/Terraria/release_extras/LaunchUtils/InstallNetFramework.sh

```diff
@@ -33,35 +36,43 @@ echo "Cleanup Complete"
 #If the install directory for this specific dotnet version doesnt exist, grab the installer script and run it.
 echo "Checking for Updates to .NET"
 if [ ! -d "$install_dir" ]; then
-	echo "Updated Required. Will now attempt downloading using official scripts. This can take up to 5 minutes"
+	echo "Updated Required. Will now attempt downloading. This can take up to 5 minutes"
 	dotnet_runtime=dotnet
 
 	if [[ "$_uname" == *"_NT"* ]]; then
-		file_download dotnet-install.ps1 https://dot.net/v1/dotnet-install.ps1
-		
-		# @TODO: Should update powershell to 4+ because required by the script (and not present by default in win 7/8)
-		powershell.exe -NoProfile -ExecutionPolicy unrestricted -File dotnet-install.ps1 -Channel "$channel" -InstallDir "$install_dir" -Runtime "$dotnet_runtime" -Version "$dotnet_version"
+		mkdir "$dotnet_dir"
+
+		# Allow for zip to be already delivered by steam win on system
+		# and placed in the root directory with this name convention:
+		dotnet_portable_archive="$root_dir/win-x64-$dotnet_version.zip"
```

Comment:
I'd use the full `dotnet-runtime-$dotnet_version-win-x64.zip` so anyone looking at it knows what it is. Also, maybe only delete it if it was auto-downloaded? Otherwise steam is likely to try and restore it inconsistently.